### PR TITLE
Feat user profile page

### PR DESCRIPTION
**Pull request by amartya-rs**
In this PR:
- added user profile page
- made user login persistent

Files to check: [review me only](https://github.com/amartya-rs/hhTube/pull/10/files/ea9cae1151765ac2a120f835d29df9845377a1d1..d817066473c5cc15b336c235734ecac499af6fb1)

Preview: https://deploy-preview-10--hh-tube.netlify.app/

### CODE REVIEW

**Review by tonymike99 — APPROVED** [review]
Hi Amartya,
Nice code. And the functionalities are working fine. 
Good to merge.

File: src/components/top-navigation/top-nav.css [hunk 1]
```diff
@@ -15,3 +15,48 @@
...
+   border: none;
+   height: 2.15rem;
+   width: 2.15rem;
+   cursor: pointer;
+}
+
+.profile-wrapper {
+   height: 4rem;
+
+   display: flex;
```
Comment (iRohitGaur) [inline src/components/top-navigation/top-nav.css:29]: suggestion: remove empty line

**Review by iRohitGaur — APPROVED** [review]
Good to merge 👍🏻

File: src/context/auth-context.js [hunk 2]
```diff
@@ -14,24 +15,31 @@ const AuthProvider = ({ children }) => {
...
    useEffect(() => {
       if (localStorage.getItem("token") !== null) {
-         authDispatch({ type: "TOGGLE_LOGIN", payload: true });
+         authDispatch({
+            type: "SET_USER_CREDENTIALS",
+            payload: getLocalData("user"),
+         });
       }
+      // eslint-disable-next-line
    }, []);
```
Comment (sudipkundu999) [inline src/context/auth-context.js]: This needs a change of logic!
You can't just say a user exists by checking if the token is null.
A token in local storage can be tampered with meaning someone can edit the token directly!
You need to verify the token to see if the user exists or not by making a call to the backend.
Check this : [verifyUser Controller](https://github.com/sudipkundu999/CobraTube/blob/development/src/backend/controllers/AuthController.js#L105)

**Review by sudipkundu999 — CHANGES_REQUESTED** [review]

File: src/context/auth-context.js (hunk 2 above)
Comment (amartya-rs) [reply src/context/auth-context.js]: Thank you very much Sudip, for pointing this out. I have fixed this issue.

File: src/components/top-navigation/top-nav.css (hunk 1 above)
Comment (amartya-rs) [reply src/components/top-navigation/top-nav.css:29]: I tend to separate out the flex properties from the other by a line.

**Review by sudipkundu999 — APPROVED** [review]
Changes look good and work as expected. 🚀 
Good work. Good to merge.

**Pull request merged**